sklearn/feature_selection/_base.py

```diff
@@ -74,9 +77,13 @@ def transform(self, X):
         X_r : array of shape [n_samples, n_selected_features]
             The input samples with only the selected features.
         """
-        tags = self._get_tags()
-        X = check_array(X, dtype=None, accept_sparse='csr',
-                        force_all_finite=not tags.get('allow_nan', True))
+        force_all_finite = not _safe_tags(self, key="allow_nan", default=True)
```

Comment:
Shouldn't this be `self._get_tags()`?

It should be fine as long as `_safe_tags(self.estimator_)` is used where needed.

Reply:
This is a `Mixin` which does not inherit from `BaseEstimator`. If a class inherit from the `Mixin` but not from `BaseEstimator`, `_get_tags` would not be defined, isn't it?

Reply:
Good point... but that Mixin is private isn't it?

Reply:
Not as far that I know. I think that we want people to inherit from `ClassifierMixin` or `RegressionMixin` to get the default score.

Reply:
Right, it's public indeed. Then maybe add a comment, because 1-year-older-me will see that and make a PR to use `_get_tags()` instead ;)

`# note: we use _safe_tags instead of _get_tags because this is a public Mixin`